shared/src/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -629,19 +630,40 @@ public void run() {
         c4ReplListener.statusChanged(c4repl, c4ReplStatus, this);
     }
 
-    private void documentEnded(boolean pushing, String docID, String revID, int flags, C4Error error, boolean trans) {
-        DocumentReplication update = new DocumentReplication((Replicator) this,
-                flags == C4Constants.C4RevisionFlags.kRevDeleted,
-                 pushing, docID, revID, flags, error, trans);
-        if (!pushing && error.getDomain() == LiteCoreDomain && error.getCode() == kC4ErrorConflict) {
-            // Conflict pulling a document -- the revision was added but app needs to resolve it:
-            Log.i(TAG, "%s: pulled conflicting version of '%s'", this, docID);
+    private void documentEnded(boolean pushing, C4DocumentEnded[] documents) {
+        ReplicatedDocument[] docs = new ReplicatedDocument[documents.length];
+        int i = 0;
+        for (C4DocumentEnded c4document: documents) {
+            boolean isAccessRemoved = false;
+            boolean isDeleted = false;
+            boolean clearError = false;
+            C4Error error = new C4Error(0,0,0);
             try {
-                this.config.getDatabase().resolveConflictInDocument(docID);
-            } catch (CouchbaseLiteException ex) {
-                Log.e(TAG, "Failed to resolveConflict: docID -> %s", ex, docID);
+                isAccessRemoved = config.getDatabase().c4db.get(c4document.getDocID(), false).accessRemoved();
+                isDeleted = config.getDatabase().c4db.get(c4document.getDocID(), false).deleted();
+            } catch (LiteCoreException e) {
+                Log.e(TAG, "C4Document does not exist: docID ->%s", c4document.getDocID());
             }
+
+            if (!pushing && c4document.getErrorDomain() == LiteCoreDomain && c4document.getErrorCode() == kC4ErrorConflict) {
+                // Conflict pulling a document -- the revision was added but app needs to resolve it:
+                Log.i(TAG, "%s: pulled conflicting version of '%s'", this, c4document.getDocID());
+                try {
+                    this.config.getDatabase().resolveConflictInDocument(c4document.getDocID());
+                } catch (CouchbaseLiteException ex) {
+                    Log.e(TAG, "Failed to resolveConflict: docID -> %s", ex, c4document.getDocID());
+                } finally {
+                    clearError = true;
```

Comment:
If the exception happens when resolving the conflict, the error shouldn't be cleared.